amp/WEB-INF/src/org/digijava/kernel/geocoding/service/GeoCodingService.java

```diff
@@ -0,0 +1,194 @@
+package org.digijava.kernel.geocoding.service;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.digijava.kernel.entity.geocoding.GeoCodedActivity;
+import org.digijava.kernel.entity.geocoding.GeoCodedField;
+import org.digijava.kernel.entity.geocoding.GeoCodedLocation;
+import org.digijava.kernel.entity.geocoding.GeoCoding;
+import org.digijava.kernel.exception.DgException;
+import org.digijava.kernel.geocoding.client.GeoCoderClient;
+import org.digijava.kernel.persistence.PersistenceManager;
+import org.digijava.module.aim.dbentity.AmpActivityVersion;
+import org.digijava.module.aim.dbentity.AmpCategoryValueLocations;
+import org.digijava.module.aim.dbentity.AmpTeamMember;
+import org.digijava.module.aim.util.ActivityUtil;
+import org.digijava.module.aim.util.LocationUtil;
+import org.digijava.module.aim.util.TeamUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Octavian Ciubotaru
+ */
+public class GeoCodingService {
+
+    private static final Logger logger = LoggerFactory.getLogger(GeoCodingService.class);
+
+    private static GeoCoderClient client = new GeoCoderClient();
+
+    public void processActivities(Set<Long> activityIds) {
+        GeoCoding geoCoding = getCurrentGeoCoding();
+
+        AmpTeamMember principal = getPrincipal();
+
+        if (geoCoding != null && !geoCoding.getTeamMember().equals(principal)) {
+            throw new GeoCodingNotAvailableException(geoCoding.getTeamMember());
+        }
+
+        if (geoCoding == null) {
+            geoCoding = new GeoCoding(principal);
+            PersistenceManager.getSession().save(geoCoding);
+        }
+
+        // TODO update locations only if changed
+        client.sendLocations(LocationUtil.getAllVisibleLocations());
+
+        List<AmpActivityVersion> activities = new ArrayList<>();
+        for (Long activityId : activityIds) {
+            try {
+                AmpActivityVersion v = ActivityUtil.loadActivity(activityId);
+                if (!v.getLocations().isEmpty()) {
```

Comment:
If the activity has a location at national level (country assigned), it shouldn't be an error.

Reply:
Ok.